### PR TITLE
Stop fetching data when pure dependencies specified

### LINKED ISSUE #2839: [BUG] xgboost on Mars will try fetching all DMatrices onto tracker machine
**Describe the bug**
xgboost on Mars will try fetching all DMatrices onto tracker machine, making the whole task stuck.

### PR DESCRIPTION
**Pull request by wjsi**
## What do these changes do?

Remove reliance of `Fetch` chunks when it is a pure dependency.

## Related issue number

Fixes #2839

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

**Review by hekaisheng — APPROVED** [review]
LGTM

File: mars/services/scheduling/worker/execution.py [hunk 1]
```diff
@@ -167,15 +167,10 @@ async def _prepare_input_data(self, subtask: Subtask, band_name: str):
         storage_api = await StorageAPI.create(
             subtask.session_id, address=self.address, band_name=band_name
         )
-        pure_dep_keys = set()
```
Comment (qinxuye) [inline mars/services/scheduling/worker/execution.py:170]: Please explain the problem why this logic does not work?

Comment (wjsi) [reply mars/services/scheduling/worker/execution.py:170]: `pure_dep_keys` is only updated but not used.

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**